src/components/Pressable/Pressable.tsx

```diff
@@ -142,9 +162,36 @@ export default function Pressable(props: PressableProps) {
   const onEndHandlingTouchesDown = useRef<(() => void) | null>(null);
   const cancelledMidPress = useRef<boolean>(false);
 
-  const touchGesture = useMemo(
+  const activateLongPress = useCallback(
+    (event: GestureTouchEvent) => {
+      if (isTouchPropagationAllowed.current === false) {
+        return;
+      }
+
+      if (hasPassedBoundsChecks.current) {
+        props.onLongPress?.(gestureTouchToPressableEvent(event));
+        isPressCallbackEnabled.current = false;
+      }
+
+      if (longPressTimeoutRef.current) {
+        clearTimeout(longPressTimeoutRef.current);
+        longPressTimeoutRef.current = null;
+      }
+    },
+    [props]
+  );
+
+  const longPressTimeoutRef = useRef<number | null>(null);
+  const longPressMinDuration =
+    (props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION) +
+    (props.unstable_pressDelay ?? 0);
+
+  const pressAndTouchGesture = useMemo(
     () =>
-      Gesture.Manual()
+      Gesture.LongPress()
+        .minDuration(Number.MAX_SAFE_INTEGER) // Stops long press from blocking native gesture
+        .maxDistance(Number.MAX_SAFE_INTEGER) // Stops long press from cancelling after set distance
```

Comment:
Why do we even need LongPress in this case? Since the actual callback is invoked by a timeout anyway, cannot it be handled by the native gesture?

Reply:
Native gesture has an unpredictable behaviour, which requires large amounts of additional code for edge case checking to synchronise between the different platforms, and this also affects touches - I checked. 
On the other hand, the core gestures as far as I see, behave identically between all the platforms, which is why i chose to use `LongPress`.

Of course, if you believe there is a good reason to really try and get rid of it in favour of either `Manual` gesture, or `Native` gesture, I can do that. I just think it may add anywhere from 50 to 150 new lines of code, and may cause some unforeseen bugs caused by newly added edge cases.

For example, when debugging the `Native` gesture on iOS, I composed a list of edge cases all of which had to be individually found through rigorous testing, and had to be uniquely addressed:
 
- inside scrollable
  - nested
    - click for <=1frame (before measure)
    - click for >1frame <200ms (before native)
    - click for >200ms <500ms (before long press)
    - click for >500ms (with long press)
  - not-nested
    - click for <=1frame (before measure)
    - click for >1frame <200ms (before native)
    - click for >200ms <500ms (before long press)
    - click for >500ms (with long press)
- not in scrollable
  - nested
    - click for >200ms <500ms (before long press)
    - click for >500ms (with long press)
  - not-nested
    - click for >1frame <200ms (before native)
    - click for >200ms <500ms (before long press)
    - click for >500ms (with long press)

I referenced this choice further in [this comment](https://github.com/software-mansion/react-native-gesture-handler/pull/2981#issuecomment-2233477313)

Reply:
During some further testing with @j-piasecki we found issues with changing `LongPress` to `Manual` or moving the logic to `Native`, and found that not much logic can be moved either way.
And so we decided to leave the code as it is for now, thus marking this as resolved.